app/models/concerns/module_settings_concern.rb

```diff
@@ -0,0 +1,35 @@
+module ModuleSettingsConcern
```

Comment:
Maybe this should be called EnableableModuleHost or something like this. The fact that a host can store settings should be declared on the module host itself.